card/src/main/java/com/adyen/checkout/card/CardDelegate.kt

```diff
@@ -30,4 +38,23 @@ abstract class CardDelegate(protected val cardConfiguration: CardConfiguration)
     abstract fun isCvcHidden(): Boolean
     abstract fun requiresInput(): Boolean
     abstract fun isHolderNameRequired(): Boolean
+
+    suspend fun fetchPublicKey(): String {
+        return if (cardConfiguration.publicKey.isNotEmpty() && CardValidationUtils.isPublicKeyValid(cardConfiguration.publicKey)) {
+            Logger.d(TAG, "returning configuration publicKey")
+            cardConfiguration.publicKey
+        } else {
+            Logger.d(TAG, "fetching publicKey from API")
+            for (retry in 1..CONNECTION_RETRIES) {
```

Comment:
Minor: you can use `repeat(CONNECTION_RETRIES) { ... }` since you don't need the index

Reply:
thanks 👍